frontend/src/main-components/CreateCardContent.js

```diff
@@ -15,27 +15,57 @@ class CreateCardContent extends Component {
       fromLang: "none",
       toLang: "none",
       folder: "",
+      uploadUrlFetched: false,
+      imageUploadUrl: "",
     };
     this.translateText = this.translateText.bind(this);
     this.fromLangSelected = this.fromLangSelected.bind(this);
     this.toLangSelected = this.toLangSelected.bind(this);
+    this.imageSelected = this.imageSelected.bind(this);
+    this.imageUploadUrl = "";
+  }
+
+  async componentDidMount() {
+    try {
+      const uploadUrl = await fetch("/upload")
+        .then((response) => { return response.text() });
+      // Throws error if the /upload servlet is not found
+      if (!uploadUrl.ok) {
+        throw Error(uploadUrl.statusText);
+      }
+      this.setState({ imageUploadUrl: uploadUrl, uploadUrlFetched: true });
+    } catch (error) {
+      alert("Refresh page to create a card");
+    }
   }
 
   /**
    * When the user has finished typing, the Google Translate
    * API is called to fetch the translated version of the text input.
    */
-  translateText(event) {
+  async translateText(event) {
+    event.persist();
+    const target = event.target.value;
```

Comment:
Is target the best name for this variable?

Reply:
done